lib/config/defaults.js

```diff
@@ -1434,6 +1434,7 @@ const applyOptimizationDefaults = (
 	D(optimization, "innerGraph", production);
 	D(optimization, "mangleExports", production);
 	D(optimization, "concatenateModules", production);
+	D(optimization, "entryIife", production);
```

Comment:
It should be `false` by default in production as it can negatively affect the build performance.

```suggestion
	D(optimization, "entryIife", !production);
```

Reply:
Yes, it will affect the performance of building. I am open to whether this option is turned on by default. Let me briefly describe the characteristics of this option.

- This optimization does not take effect in all building scenarios. Generally, it usually takes effect when building a library.
- Unfortunately, we have not run any benchmarks here, but what can be determined is that the impact on build performance will grow with the number of entry modules and chunks modules
- It's a trade-off between build efficiency and execution efficiency.

Reply:
How do we acheive the old behavior before #18348? by `entryIife: true` or `entryIife : false`?

Reply:
As per the description in schema I think it should be `entryIife: true`, I got confused with the option's name.

Reply:
> right now, how do we achieve the old behavior before https://github.com/webpack/webpack/pull/18348? by entryIife: true or entryIife : false?

By `entryIife: false`, `true` means enable the optimization.

Reply:
@alexander-akait What do you think here, given that there are not many users complaining about IIFE (I searched issue, please correct me if I don't find it), I think it is also fine to shut down IIFE at production. We can turn this option on in `'modern-module'` without any problems.

Reply:
Yeah, no issues, but we have todo

```js
// TODO check globals and top-level declarations of other entries and chunk modules
// to make a better decision
```

I would consider this part of our todo, not only for `modern-module`